src/d3d12/d3d12_renderer.cpp

```diff
@@ -180,7 +176,7 @@ namespace wr
 		d3d12::Signal(m_fences[frame_idx], m_direct_queue);
 	}
 
-	void D3D12RenderSystem::RenderSceneGraph(SceneGraph const & scene_graph)
+	void D3D12RenderSystem::RenderSceneGraph(SceneGraph& scene_graph)
```

Comment:
This should really be const. You wouldn't expect changes in a render func.

Reply:
Not possible; SceneGraph::Optimize sets up the batches inside of RenderSceneGraph -> RenderMeshBatches; which requires write access.